### PR TITLE
scm: reset tracked files after every operation

### LINKED ISSUE #1475: Deduplicate files in `git add` command suggestion
0.23.2

When running a `dvc run` with multiple outputs it prints `git add` with `.gitignore` repeated multiple times.

ss

### PR DESCRIPTION
**Pull request by mroutis**
Fix #1475

* [x] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing)?

* [x] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----

### CODE REVIEW

**Comment by efiop** [comment]
@mroutis Now it doesn't print anything with your patch. E.g.
```
(3.7.0-dvc) ➜  dvc git:(fix-1475) ✗ dvc run -o foo -o bar 'echo foo > foo && echo bar > bar'
Running command:                                                                            
        echo foo > foo && echo bar > bar                                                    
Saving 'foo' to '.dvc/cache/d3/b07384d113edec49eaa6238ad5ff00'.                             
Saving 'bar' to '.dvc/cache/c1/57a79031e1c40f85931829bc5fc552'.                             
Saving information to 'foo.dvc'.                                                            
```

**Comment by mroutis** [comment]
woah, that was awful :sweat_smile: misplaced the new line

File: dvc/scm/git/__init__.py [hunk 1]
```diff
@@ -253,7 +256,9 @@ def remind_to_track(self):
             "\n"
             "To track the changes with git, run:\n"
             "\n"
-            "\tgit add {files}".format(files=" ".join(self.files_to_track))
+            "\tgit add {files}".format(
+                files=" ".join(set(self.files_to_track))
```
Comment (efiop) [inline dvc/scm/git/__init__.py]: so maybe we should actually make self.files_to_track a set?

Comment (mroutis) [reply dvc/scm/git/__init__.py]: @efiop, can you review this again, please?

File: dvc/scm/git/__init__.py [hunk 2]
```diff
@@ -257,7 +260,7 @@ def remind_to_track(self):
         )
 
     def track_file(self, path):
-        self.files_to_track.append(path)
+        self.files_to_track.update(set(path))
```
Comment (efiop) [inline dvc/scm/git/__init__.py]: Why not just?
```suggestion
        self.files_to_track.add(path)
```

**Pull request merged**